shared/tinyusb/mp_usbd_cdc.c

```diff
@@ -162,6 +172,15 @@ tud_cdc_line_state_cb
         tud_sof_cb_enable(true);
     }
     #endif
+    #if MICROPY_HW_ESP_AUTOMATIC_BOOTLOADER
```

Comment:
I reckon this new config option should be called `MICROPY_HW_USB_CDC_DTR_RTS_BOOTLOADER`, to indicate that it's part of the USB CDC code.

And then put a default value in `shared/tinyusb/mp_usbd_cdc.h` of 0 (disabled) with a comment that this should only be used on ESP32 boards.